### ISSUE TITLE
Query: Better exception message when invalid expression in Include

### ISSUE THREAD

**Issue by rowanmiller** [aspnet/EntityFrameworkCore#1883]
See this thread https://github.com/aspnet/EntityFramework/issues/1833
We should provide a clearer error stating that only navigation paths are allowed at the moment

**Comment by smitpatel** [comment]
@maumar - What would be current exception message?

**Comment by maumar** [comment]
Fix for #3910 should address this. New exception is (will double check that the scenario in the bug is affected)

```
Lambda expression '{includeLambdaExpression}' is not valid for Include/ThenInclude result operator. The expression should represent a property access: 't => t.MyProperty'. Specify lambda parameter explicitly to access navigations on derived types: '(Derived t) => t.MyProperty'. For more information on including related data, see
```
